snapshot/kaniko_snapshot_unix_test.go

```diff
@@ -0,0 +1,157 @@
+// +build linux darwin
+
+package snapshot_test
+
+import (
+	"archive/tar"
+	"bytes"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/sclevine/spec"
+
+	"github.com/buildpacks/lifecycle/snapshot"
+	h "github.com/buildpacks/lifecycle/testhelpers"
+)
+
+func TestKanikoSnapshotter(t *testing.T) {
+	spec.Run(t, "Test Image", testKanikoSnapshotter)
+}
+
+func testKanikoSnapshotter(t *testing.T, when spec.G, it spec.S) {
+	var (
+		snapshotter *snapshot.KanikoSnapshotter
+		tmpDir      string
+	)
+
+	it.Before(func() {
+		// Using the default tmp dir causes kaniko to go haywire for some reason
```

Comment:
I actually don't know. @jkutner might remember (he originally wrote this bit)

Reply:
I took a quick look just now. The problem is that the the test times out while getting the initial kaniko snapshot when give `""` for the directory to create the temporary directory. I believe the reason this happens is because macOS uses symlinks for the default tmp directory path. 

```
» echo $TMPDIR
/var/folders/cx/xmrdk7sx6x5gcv0gl951jhx00000gp/T/
» ls -al /var
lrwxr-xr-x 1 root admin 11 Jan 23  2020 /var -> private/var
```

Kaniko has some code when walking the filesystem to check ancestors for symlinks. I think kaniko can't get out of the loop of it trying to resolve the symlinks properly. Maybe a bug in kaniko?

```
time="2020-10-07T12:58:58-05:00" level=info msg="Taking snapshot of full filesystem..."
time="2020-10-07T12:58:59-05:00" level=trace msg="Resolving paths [/var/folders/cx/xmrdk7sx6x5gcv0gl951jhx00000gp/T/kaniko-snapshotter883391310 /var/folders/cx/xmrdk7sx6x5gcv0gl951jhx00000gp/T/kaniko-snapshotter883391310/cnb /var/folders/cx/xmrdk7sx6x5gcv0gl951jhx00000gp/T/kaniko-snapshotter883391310/cnb/privatefile /var/folders/cx/xmrdk7sx6x5gcv0gl951jhx00000gp/T/kaniko-snapshotter883391310/layers /var/folders/cx/xmrdk7sx6x5gcv0gl951jhx00000gp/T/kaniko-snapshotter883391310/layers/privatefile /var/folders/cx/xmrdk7sx6x5gcv0gl951jhx00000gp/T/kaniko-snapshotter883391310/bin /var/folders/cx/xmrdk7sx6x5gcv0gl951jhx00000gp/T/kaniko-snapshotter883391310/bin/file-not-to-change /var/folders/cx/xmrdk7sx6x5gcv0gl951jhx00000gp/T/kaniko-snapshotter883391310/file-not-to-change /var/folders/cx/xmrdk7sx6x5gcv0gl951jhx00000gp/T/kaniko-snapshotter883391310/file-to-change /var/folders/cx/xmrdk7sx6x5gcv0gl951jhx00000gp/T/kaniko-snapshotter883391310/file-to-delete]"
time="2020-10-07T12:58:59-05:00" level=trace msg="updated link /var/folders/cx/xmrdk7sx6x5gcv0gl951jhx00000gp/T/kaniko-snapshotter883391310/cnb/privatefile to /var/folders/cx/xmrdk7sx6x5gcv0gl951jhx00000gp/T/kaniko-snapshotter883391310/cnb"
time="2020-10-07T12:58:59-05:00" level=trace msg="updated link /var/folders/cx/xmrdk7sx6x5gcv0gl951jhx00000gp/T/kaniko-snapshotter883391310/layers/privatefile to /var/folders/cx/xmrdk7sx6x5gcv0gl951jhx00000gp/T/kaniko-snapshotter883391310/layers"
time="2020-10-07T12:58:59-05:00" level=trace msg="updated link /var/folders/cx/xmrdk7sx6x5gcv0gl951jhx00000gp/T/kaniko-snapshotter883391310/bin/file-not-to-change to /var/folders/cx/xmrdk7sx6x5gcv0gl951jhx00000gp/T/kaniko-snapshotter883391310/bin"
panic: test timed out after 30s
```

Reply:
Ah... over in `pack` we've used `filepath.EvalSymlinks(tmpDir)` to resolve issues with volume mounts that I think stem from the same problem. Might be worth giving it a try...